packages/repository/src/__tests__/unit/repositories/legacy-juggler-bridge.unit.ts

```diff
@@ -416,4 +418,21 @@ describe('DefaultCrudRepository', () => {
       'execute() must be implemented by the connector',
     );
   });
+
+  it('throws an error when beginTransaction() not implemented', async () => {
+    const repo = new DefaultCrudRepository(Note, ds);
+    await expect(repo.beginTransaction({})).to.be.rejectedWith(
+      'beginTransaction must be function implemented by the connector',
```

Comment:
`beginTransaction must be implemented`?

Reply:
How about this:

```suggestion
      'Cannot start a new transaction - "memory" connector does not support transactions.'
```

(Replace `memory` with the actual connector name.)

Reply:
This is coming from `loopback-connector`. See https://github.com/strongloop/loopback-connector/blob/master/lib/transaction.js#L96-L97. Perhaps we can improve the message there?

Reply:
Yes please, let's improve the message there. Ideally, the error object should have also a `code` property set to a reasonably-distinct string, e.g. `'TRANSACTIONS_NOT_SUPPORTED'`. That will allow us to avoid brittle string check and verify just the error `code` in this test.

Anyhow, I am fine to leave this improvement out of scope of this pull request if you prefer.